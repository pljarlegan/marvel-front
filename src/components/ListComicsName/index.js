import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListSubheader from '@material-ui/core/ListSubheader';


const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

export default function ListComicsName({ title, items }) {
  const classes = useStyles();

  return (
    <List subheader={items.length >0 ? <ListSubheader>{title}</ListSubheader> : null} className={classes.root}>
      { items.map(item => <ListItem key={item.name}>{item.name}</ListItem>)}
    </List>
  );
}
