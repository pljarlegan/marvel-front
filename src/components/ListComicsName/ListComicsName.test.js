import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ListComicsName from './index'

configure({ adapter: new Adapter() });

describe('test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListComicsName title={"bob"} items={[]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Matches the snapshot title 'bob', 0 items", () => {
    const wrapper = shallow(<ListComicsName title={"bob"} items={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Matches the snapshot title 'bob', 2 items", () => {
    const wrapper = shallow(<ListComicsName title={"bob"} items={[{name: "a"}, {name: "b"}]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});


