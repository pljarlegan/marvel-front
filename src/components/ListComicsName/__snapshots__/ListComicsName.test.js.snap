// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches the snapshot title 'bob', 0 items 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-8"
  subheader={null}
/>
`;

exports[`test Matches the snapshot title 'bob', 2 items 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-8"
  subheader={
    <ForwardRef(WithStyles)>
      bob
    </ForwardRef(WithStyles)>
  }
>
  <WithStyles(ForwardRef(ListItem))
    key="a"
  >
    a
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))
    key="b"
  >
    b
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;
